src/gcode-streamer_test.cc

```diff
@@ -0,0 +1,208 @@
+/* -*- mode: c++; c-basic-offset: 2; indent-tabs-mode: nil; -*-
+ * Test for pru motion queue.
+ *
+ * We simulate the pru hardware and test if the absolute position is correctly
+ * retrieved.
+ *
+ */
+#include "motion-queue.h"
+
+#include <stdio.h>
+#include <string.h>
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include "common/fd-mux.h"
+#include "common/logging.h"
+#include "gcode-parser/gcode-parser.h"
+#include "gcode-streamer.h"
+
+class MockStream {
+public:
+
+  MockStream() : fd_{-1, -1} {}
+  ~MockStream() { ClientDisconnect(); }
+
+  int ClientConnect() {
+    ClientDisconnect();
+    if (pipe(fd_) < 0) {
+      perror("pipe()");
+      exit(1);
+    }
+    return fd_[0];
+  }
+
+  int NewData(const char *data) {
+    // Feed data in the pipe
+    return write(fd_[1], data, strlen(data));
+  }
+
+  void ClientDisconnect() { close(fd_[1]); }
+
+private:
```

Comment:
Tip: I'd add some constants (e.g. an enum) to contain the numbers of the ends of the pipe
```
enum { RECEIVING_FD, SENDING_FD };
```

Then above, it is much easier to read what we're using via the symbolic names e.g.
```
  int SendData(const char *data) {
      return write(fd_[SENDING_FD], data, strlen(data));
  }
```